### PR TITLE
Gjoshi/anonymous user handling

### PR DESCRIPTION
**Pull request by dkjellin**
Mostly Gaurav's work, to handle unsecured instances (such as the one spun up via hpi:run),

### CODE REVIEW

**Review by goravjoshi — APPROVED** [review]
Looks good to me. There was one thing that could trouble us. It is unrelated to anonymous. If the default permission is allow read to anonymous, we need a way to force login through authorization screen. Otherwise, the token would not be associated with a user. This was something I found out while testing. May be @mhenschke-atl knows how to force login through jelly files.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/auth/OAuth1aRequestFilter.java [hunk 1]
```diff
@@ -57,18 +59,21 @@
...
+    private final JenkinsProvider jenkinsProvider;
 
     @Inject
     public OAuth1aRequestFilter(ServiceProviderConsumerStore consumerStore,
                                 ServiceProviderTokenStore tokenStore,
                                 OAuthValidator validator,
                                 Clock clock,
-                                TrustedUnderlyingSystemAuthorizerFilter authorizerFilter) {
+                                TrustedUnderlyingSystemAuthorizerFilter authorizerFilter,
+                                JenkinsProvider jenkinsProvider) {
```
[the remark this answers is not in the document]
Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/auth/OAuth1aRequestFilter.java:70]: Right, so you're proposing we change this to something more agnostic? I can do that if you think it is worth it.

**Comment by mhenschke-atl** [comment]
Re. Anonymous read, yes we set permissions on the screen- if users don't meet them, they are prompted to log in automatically. Right now they're set to app read. You can set this in the `AuthorizeConfirmationConfig` jelly.

**Review by alibk0rd — APPROVED** [review]

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/auth/OAuth1aRequestFilter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (goravjoshi) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/auth/OAuth1aRequestFilter.java:70]: Yes. I think it is worth it. So far there is no dependency arrow pointing to Jenkins stuff. I have mentioned that in the OAuth [documentation](https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/blob/master/docs/oauth_provider.md#applink-module) as well that there is a clear boundaries for this.

**Review by mhenschke-atl — APPROVED** [review]

**Pull request merged**